### DISCUSSION TITLE
Select a recipe's slot inside another slot in the recipe definition

### DISCUSSION THREAD

**Discussion by RemyMachado** [chakra-ui/panda#2331] [Q&A] [answered]
I was wondering if there was a way to select a slot inside another.
Here's what I tried and I think it illustrates well what I'm trying to do:

```typescript
defineSlotRecipe({
  slots: ['slot1', 'slot2'],
  base: {
    slot1: {
       '& > slot2': {
         color: '2'
      }
    },

    slot2: {
      color: '1'
    },
  }
}
```

I also tried using the class selector, `& > .slot2` (but it's obviously not working since the slot names aren't equal to their final classname)

**Comment by astahmer** [comment, marked as answer]
hey, recipes have deterministic className generation based on this: `${className}__${slotName}` 

```ts
const button = defineRecipe({
  className: 'btn',
  slots: ['root', 'text'],
  base: {
    root: {
      bg: 'blue.500',
      _hover: {
        // v--- 🎯 this will target the `text` slot
        '& .btn__text': {
          color: 'white',
        },
      },
    },
  },
})
```

also, I just added a PR to do the same with `sva` https://github.com/chakra-ui/panda/pull/2348

**Reply by p-dubovitsky** [reply]
It doesn‘t work if the hash is enabled
